xcdat/dataset.py

```diff
@@ -315,9 +317,15 @@ def decode_non_cf_time(dataset: xr.Dataset) -> xr.Dataset:
     except ValueError:
         return ds
 
-    ref_date = pd.to_datetime(ref_date)
+    ref_date = pd.to_datetime(ref_date).to_pydatetime()
+
+    if units == "months":
+        data = [ref_date + rd.relativedelta(months=offset) for offset in time.data]
+        data = [cftime.datetime(t.year, t.month, t.day) for t in data]
```

Comment:
I ended up using `cftime` because xarray needs to use cftime in some instances, e.g., `ds = xr.open_mfdataset('/p/css03/esgf_publish/CMIP6/ScenarioMIP/MRI/MRI-ESM2-0/ssp585/r1i1p1f1/Amon/ta/gn/v20191108/*nc', use_cftime=True)`. 

But this breaks our unit tests for the non-cf-compliant data. I'm not sure if we can do other things with cftime (e.g., temporal averaging and things like that). I think those functions should work, but I do not know...

Reply:
A more clear example of this is the following, where xarray automatically decodes with cftime:

```
fn = '/p/css03/esgf_publish/CMIP6/ScenarioMIP/MRI/MRI-ESM2-0/ssp585/r1i1p1f1/Amon/ta/gn/v20191108/ta_Amon_MRI-ESM2-0_ssp585_r1i1p1f1_gn_225101-230012.nc'
ds = xr.open_dataset(fn)  
```

> /home/pochedley1/bin/anaconda3/envs/xcdat_dev/lib/python3.9/site-packages/xarray/coding/times.py:673: SerializationWarning: Unable to decode time axis into full numpy.datetime64 objects, continuing using cftime.datetime objects instead, reason: dates out of range                                                                   
>   dtype = _decode_cf_datetime_dtype(data, units, calendar, self.use_cftime)
> /home/pochedley1/bin/anaconda3/envs/xcdat_dev/lib/python3.9/site-packages/xarray/coding/times.py:673: SerializationWarning: Unable to decode time axis into full numpy.datetime64 objects, continuing using cftime.datetime objects instead, reason: dates out of range
>   dtype = _decode_cf_datetime_dtype(data, units, calendar, self.use_cftime)
> /home/pochedley1/bin/anaconda3/envs/xcdat_dev/lib/python3.9/site-packages/xarray/core/indexing.py:423: SerializationWarning: Unable to decode time axis into full numpy.datetime64 objects, continuing using cftime.datetime objects instead, reason: dates out of range
>   return np.asarray(array[self.key], dtype=None)